modules/core/src/lib/effect-manager.ts

```diff
@@ -16,9 +17,24 @@ export default class EffectManager {
     this._setEffects([]);
   }
 
-  addDefaultEffect(effect: Effect) {
-    if (!this._defaultEffects.find(e => e.constructor === effect.constructor)) {
-      this._defaultEffects.push(effect);
+  // TODO - better Effect API, may be cleaner to add `order` to `Effect` class
+  /**
+   * Register a new default effect, i.e. an effect present regardless of user supplied props.effects
+   */
+  addDefaultEffect(
+    /** Effect instance. The same effect can only be added once. */
+    effect: Effect,
+    /** Used to sort effects. In case of conflict, effects will be used in the order of registration. */
+    order: number = Infinity
+  ) {
+    const defaultEffects = this._defaultEffects;
+    if (!defaultEffects.find(e => e.effect.constructor === effect.constructor)) {
```

Comment:
Retrospective design nit: I get the constructor check but it does feel weird enough that it makes me want to go back and rethink the API. E.g. could effects at this level be object descriptors instead of classes. For another day. FWIW I have been iterating on a similar separation in luma.gl/shadertools for the shadermodule definitions/instances. Maybe a reusable pattern will emerge.